File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>mp;
        int n=nums.size();
        for(int i=0;i<n;i++){
           int rem=target-nums[i];
           if(mp.find(rem)!=mp.end()){
              return {i,mp[rem]};
           }
           mp[nums[i]]=i;
        }
        return {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
        // for(int i=0;i<n;i++){
        //    for(int j=i+1;j<n;j++){
        //      if(nums[i]+nums[j]==target){
        //         return new int[]{i,j};
        //      }
        //    }
        // }
        // return new int[]{-1,-1};
        Map<Integer,Integer>mp=new HashMap<>();
        for(int i=0;i<n;i++){
            int rem=target-nums[i];
           if(mp.containsKey(target-nums[i])){
              return new int[]{mp.get(target-nums[i]),i};
           }
           mp.put(nums[i],i);
        }
        return new int[]{-1,-1};
    }
}